services/vision/colordetector/color_detector.go

```diff
@@ -0,0 +1,54 @@
+// Package colordetector uses a heuristic based on hue and connected components to create
+// bounding boxes around objects of a specified color.
+package colordetector
+
+import (
+	"context"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/utils"
+	objdet "go.viam.com/rdk/vision/objectdetection"
+)
+
+var model = resource.NewDefaultModel("color_detector")
+
+func init() {
+	resource.RegisterService(vision.Subtype, model, resource.Registration[vision.Service, *objdet.ColorDetectorConfig]{
+		DeprecatedRobotConstructor: func(ctx context.Context, r any, c resource.Config, logger golog.Logger) (vision.Service, error) {
```

Comment:
Can you please file a ticket to remove this used of DeprecatedRobotConstructor?

Reply:
https://viam.atlassian.net/browse/RSDK-2808